File: NOVEMBER-2025/17-11-2025/1437.erl
-spec k_length_apart(Nums :: [integer()], K :: integer()) -> boolean().
k_length_apart(Nums, K) ->
    k_length_apart(Nums, K, -1, 0).

k_length_apart([], _K, _Prev, _I) ->
    true;

k_length_apart([1 | Rest], K, Prev, I) ->
    case (Prev /= -1) andalso ((I - Prev - 1) < K) of
        true -> false;
        false -> k_length_apart(Rest, K, I, I + 1)
    end;

k_length_apart([_ | Rest], K, Prev, I) ->
    k_length_apart(Rest, K, Prev, I + 1).

File: NOVEMBER-2025/17-11-2025/1437.ex
defmodule Solution do
  @spec k_length_apart(nums :: [integer], k :: integer) :: boolean
  def k_length_apart(nums, k) do
    do_check(nums, k, -1, 0)
  end

  defp do_check([], _k, _prev, _i), do: true

  defp do_check([1 | rest], k, prev, i) do
    if prev != -1 and (i - prev - 1) < k do
      false
    else
      do_check(rest, k, i, i + 1)
    end
  end

  defp do_check([_ | rest], k, prev, i) do
    do_check(rest, k, prev, i + 1)
  end
end